march_joint_inertia_controller/include/march_joint_inertia_controller/inertia_estimator.h

```diff
@@ -0,0 +1,145 @@
+// Copyright 2020 Project March.
+
+#ifndef MARCH_WS_INERTIA_ESTIMATOR_H
+#define MARCH_WS_INERTIA_ESTIMATOR_H
+
+#include <hardware_interface/joint_command_interface.h>
+#include <ros/ros.h>
+#include <trajectory_interface/quintic_spline_segment.h>
+#include <urdf/model.h>
+
+/**
+ * \brief determines the absolute value of vector a and stores it into b so that a remains unchanged
+ *
+ * @param[in] a input vector with values that need to be converted to absolutes
+ * @param[out] b output vector with copies of the absolute values of a.
+ */
+std::vector<double> absolute(const std::vector<double>& a);
+
+/**
+ * \brief determiness the mean value of the given vector
+ *
+ * @param[in] a input vector from which the mean gets calculated
+ * @returns the mean value of a
+ */
+double mean(const std::vector<double>& a);
+
+/**
+ * \brief Class that bundles functionality to estimate inertia on a Revolute Joint.
+ *
+ * @param[in] joint the jointhandle used for obtaining current speed and effort.
+ * @param[in] lambda sets the lambda used in the RLS filter defaulted to 0.96
+ * @param[in] acc_size the size of the vector wherein the calculated acceleration is stored. This vector length is also
+ *  used in determining the length of the velocity vector and the filtered acceleration vector. It is of specific
+ *  size to calculate the alpha value. Defaulted to 12.
+ */
+class InertiaEstimator
+{
+public:
+  /**
+   * \brief Default constructor
+   */
+  InertiaEstimator(double lambda = 0.96, size_t acc_size = 12);
+
+  double getAcceleration(unsigned int index);
+
+  void setLambda(double lambda);
+  void setAcc_size(size_t acc_size);
+  void setNodeHandle(ros::NodeHandle& nh);
+  /**
+   * \brief Initialize the publisher with a name
+   */
+  void configurePublisher(const std::string& name);
+  void publishInertia();
+
+  /**
+   * \brief Applies the Butterworth filter over the last two samples and returns the resulting filtered value
+   */
+  void apply_butter();
+
+  /**
+   * \brief Estimate the inertia using the acceleration and torque
+   */
+  void inertia_estimate();
+  /**
+   * \brief Calculate a discrete derivative of the speed measurements
+   */
+  double discrete_speed_derivative(const ros::Duration& period);
+  /**
+   * \brief Calculate the alpha coefficient for the inertia estimate
+   */
+  double alpha_calculation();
+  /**
+   * \brief Calculate the inertia gain for the inertia estimate
+   */
+  double gain_calculation();
+  /**
+   * \brief Calculate the correlation coefficient of the acceleration buffer
+   */
+  void correlation_calculation();
+  /**
+   * \brief Calculate the vibration based on the acceleration
+   */
+  double vibration_calculation();
+
+  /**
+   * \brief Fill the rolling buffers with corresponding values.
+   */
+  void fill_buffers(double velocity, double effort, const ros::Duration& period);
+
+  /**
+   * \brief Calculate the initial correlation coefficient
+   */
+  void init_p(unsigned int samples);
```

Comment:
I only noticed this now, but we like to use camelCase naming for class methods. You even use them in this class as well.

Reply:
fixed it